storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -310,7 +310,7 @@ private <T> T loadIndexFile(File file, RemoteLogSegmentMetadata remoteLogSegment
         if (Files.exists(indexFile.toPath())) {
             try {
                 index = readIndex.apply(indexFile);
-            } catch (CorruptRecordException ex) {
```

Comment:
Can `CorruptRecordException` be thrown in some scenario?

Reply:
@Hangleton  The  RemoteIndexCache  caches only index files which needs to be checked for corruption. There is no such record concept in RemoteIndexCache. 
@divijvaidya  @satishd can add more details here.

Reply:
Thanks, I wanted to confirm that it is indeed the case although we are dealing with indexes here. The rationale being that `CorruptRecordException` may have been introduced purposefully to cover a use case. Was this not covered by a test in an initial PR, and if it was, did the test make an incorrect assumption, or exercised an invalid scenario?

Reply:
> Was this not covered by a test in the initial PR

@Hangleton no we don't have a test case to execute this code. In fact, the bug fixed this PR was discovered while working on the task [1] to add more unit test for remote index cache.

[1] https://issues.apache.org/jira/browse/KAFKA-15169